Added shortcode support

Codencode opened the pull request:
Added shortcode support in:
- Product: description and description_short fields
- Category: description field
- CMS: content field
- Manufacturer: description field

In each of these fields, one or more shortcodes can be added to the entities: product, category, CMS, manufacturer

<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project!

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/#pull-requests

For type and category see:
https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#type--category
------------------------------------------------------------------------------>

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 9.0.x
| Description?      | Added shortcode support for product, cateogry, cms page, cms category, manufacturer
| Type?             | new feature
| Category?         | FO, BO 
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Access one of the entities that support shortcodes, add a shortcode, and verify that the link is displayed correctly in the front office..
| UI Tests          | 
| Fixed issue or discussion?     | Idea #38108
| Related PRs       | #38212
| Sponsor company   | Codencode

ps-jarvis commented:
<!-- PR_TABLE_DESCRIPTION_ERROR -->
Hi, thanks for this contribution!

I found some issues with the Pull Request description:

* The `category` should be one of these: `FO`, `BO`, `CO`, `IN`, `WS`, `TE`, `LO`, `ME` or `PM`. ([Read explanation](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#branch))

Would you mind having a look at it? This will help us understand how interesting your contribution is, thank you very much!
### About linked issues
Please consider opening an issue before submitting a Pull Request:
* If it's a bug fix, it helps maintainers verify that the bug is effectively due to a defect in the code, and that it hasn't been fixed already.
* It can help trigger a discussion about the best implementation path before a single line of code is written.
* It may lead the Core Product team to mark that issue as a priority, further attracting the maintainers' attention.

(Note: this is an automated message, but answering it will reach a real human)

Codencode committed:
Added shortcode support

Added shortcode support in:
- Product: description and description_short fields
- Category: description field
- CMS: content field
- Manufacturer: description field

In each of these fields, one or more shortcodes can be added to the entities: product, category, CMS, manufacturer

Touxten commented:
:'(
